extension-localization/src/main/java/com/mapbox/maps/extension/localization/Localization.kt

```diff
@@ -52,6 +75,37 @@ internal fun setMapLanguage(locale: Locale, style: StyleInterface, layerIds: Lis
     }
 }
 
+fun checkServerSideLocalizationSet(listener: (Boolean) -> Unit) {
+  // create worker thread to run settingsInterface methods.
+  val workerThread = HandlerThread(
+    "SETTINGS_THREAD", THREAD_PRIORITY_DEFAULT
+  ).apply { start() }
+
+  val workerHandler = Handler(workerThread.looper)
```

Comment:
Is thread needed here? Did we measure how much time it takes to get 2 strings from settings?

Reply:
Settings interface methods cannot be called from main thread, so we are executing it on a worker thread.

> Did we measure how much time it takes to get 2 strings from settings?

haven't checked, but shouldn't take much. since its a datastore implementation.

Reply:
> Settings interface methods cannot be called from main thread

Why not?

Reply:
this is how it was designed in common... they implemented a async operation as blocking so imo it shouldn't be executed on main thread